### PR TITLE
Worker thread - member status

### PR DESCRIPTION
**Pull request by hthompson6**
Spawns a thread on driver init that updates the operating status of members.

### CODE REVIEW

**Comment by hthompson6** [comment]
check

File: a10_neutron_lbaas/v2/handler_lb.py [hunk 1]
```diff
@@ -86,6 +86,16 @@ def stats(self, context, lb):
                     "total_connections": 0
                 }
 
+    def oper(self, context, lb, lb_db, model_type):
```
Comment (dougwig) [inline a10_neutron_lbaas/v2/handler_lb.py]: this should be _oper, but let's go one step further and make that name more self-documenting. I'd suggest "_update_operating_status".

File: a10_neutron_lbaas/v2/handler_member.py [hunk 2]
```diff
@@ -145,3 +145,18 @@ def stats(self, context, member):
             pass
 
         return retval
+
+    def oper(self, context, member, lb_db, model_type, pool):
```
Comment (dougwig) [inline a10_neutron_lbaas/v2/handler_member.py]: same as above.

File: a10_neutron_lbaas/worker/status_check.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#    under the License.
+from a10_neutron_lbaas.v2 import handler_member
+
+from neutron import context as ncontext
+from neutron_lbaas.db.loadbalancer.loadbalancer_dbv2 import LoadBalancerPluginDbv2
+from neutron_lbaas.db.loadbalancer import models
+
+
+import logging
+
```
Comment (dougwig) [inline a10_neutron_lbaas/worker/status_check.py]: python import order is three blocks, each alphabetical:

system

third-party

self

File: a10_neutron_lbaas/worker/status_check.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+logging.basicConfig()
+LOG = logging.getLogger(__name__)
+
+
+def status_update(a10_driver):
```
Comment (dougwig) [inline a10_neutron_lbaas/worker/status_check.py]: What about lbaas v1?

And these neutron imports should be via the neutron_ops module in both v1 and v2.

File: test-requirements.txt [hunk 5]
```diff
@@ -8,4 +8,4 @@ nose-exclude
 pymysql
 # TODO(dougwig) -- can we remove this?
 -e git+https://github.com/openstack/neutron#egg=neutron
-
+-e git+https://github.com/openstack/neutron-lbaas#egg=neutron-lbaas
```
Comment (dougwig) [inline test-requirements.txt]: Nope, mock this out, we don't need to call down that far for a unit test.

**Comment by dougwig** [comment]
-1, feedback inline.

File: a10_neutron_lbaas/worker/main.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        self.plugin = self.a10_driver.openstack_driver.plugin
+        self.queue = kwargs.get('queue')
+
+    def run(self):
+        LOG.info("A10 worker thread, starting")
+
+        while True:
+            LOG.info("A10 worker, idling")
+            status_check.status_update(self.a10_driver)
+            time.sleep(10)
```
Comment (dougwig) [inline a10_neutron_lbaas/worker/main.py]: This is not honoring the config value for how long to sleep.

**Comment by dougwig** [comment]
I don't see any unit tests.

**Pull request closed without merging**